contribs/github-bot/client/client.go

```diff
@@ -0,0 +1,255 @@
+		}
+		return newComment
+	}
+
+	comment.Body = &body
+	editComment, _, err := gh.Client.Issues.EditComment(
+		gh.Ctx,
+		gh.Owner,
+		gh.Repo,
+		comment.GetID(),
+		comment,
+	)
+	if err != nil {
+		gh.Logger.Errorf("Unable to edit bot comment with ID %d: %v", comment.GetID(), err)
+		return nil
+	}
+	return editComment
+}
+
+func (gh *GitHub) ListTeamMembers(team string) []*github.User {
+	var (
+		allMembers []*github.User
+		opts       = &github.TeamListTeamMembersOptions{
+			ListOptions: github.ListOptions{
+				PerPage: PageSize,
+			},
+		}
+	)
+
+	for {
+		members, response, err := gh.Client.Teams.ListTeamMembersBySlug(
+			gh.Ctx,
+			gh.Owner,
+			team,
+			opts,
+		)
+		if err != nil {
+			gh.Logger.Errorf("Unable to list members for team %s: %v", team, err)
+			return nil
+		}
+
+		allMembers = append(allMembers, members...)
+
+		if response.NextPage == 0 {
+			break
+		}
+		opts.Page = response.NextPage
+	}
+
+	return allMembers
+}
+
+func (gh *GitHub) IsUserInTeams(user string, teams []string) bool {
+	for _, team := range teams {
+		for _, member := range gh.ListTeamMembers(team) {
+			if member.GetLogin() == user {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func (gh *GitHub) ListPrReviewers(prNum int) *github.Reviewers {
+	var (
+		allReviewers = &github.Reviewers{}
+		opts         = &github.ListOptions{
+			PerPage: PageSize,
+		}
+	)
+
+	for {
+		reviewers, response, err := gh.Client.PullRequests.ListReviewers(
+			gh.Ctx,
+			gh.Owner,
+			gh.Repo,
+			prNum,
+			opts,
+		)
+		if err != nil {
+			gh.Logger.Errorf("Unable to list reviewers for PR %d: %v", prNum, err)
+			return nil
+		}
+
+		allReviewers.Teams = append(allReviewers.Teams, reviewers.Teams...)
+		allReviewers.Users = append(allReviewers.Users, reviewers.Users...)
+
+		if response.NextPage == 0 {
+			break
+		}
+		opts.Page = response.NextPage
+	}
+
+	return allReviewers
+}
+
+func (gh *GitHub) ListPrReviews(prNum int) []*github.PullRequestReview {
+	var (
+		allReviews []*github.PullRequestReview
+		opts       = &github.ListOptions{
+			PerPage: PageSize,
+		}
+	)
+
+	for {
+		reviews, response, err := gh.Client.PullRequests.ListReviews(
+			gh.Ctx,
+			gh.Owner,
+			gh.Repo,
+			prNum,
+			opts,
+		)
+		if err != nil {
+			gh.Logger.Errorf("Unable to list reviews for PR %d: %v", prNum, err)
+			return nil
+		}
+
+		allReviews = append(allReviews, reviews...)
+
+		if response.NextPage == 0 {
+			break
+		}
+		opts.Page = response.NextPage
+	}
+
+	return allReviews
+}
+
+func (gh *GitHub) Close() {
+	if gh.cancel != nil {
+		gh.cancel()
+	}
+}
+
+func New(params *p.Params) *GitHub {
+	gh := &GitHub{
+		Owner:  params.Owner,
+		Repo:   params.Repo,
+		DryRun: params.DryRun,
+	}
+
+	// Detect if the current process was launched by a GitHub Action and return
+	// a logger suitable for terminal output or the GitHub Actions web interface
+	gh.Logger = logger.NewLogger(params.Verbose)
+
+	// Create context with timeout if specified in the parameters
```

Comment:
Initially, I was just discarding the cancelFn, which honestly serves no purpose (since this program runs in one go from start to finish, most functions only execute once in a well-defined order, no service to start or stop, etc.). I think it was much cleaner this way.
I then added the Close function because it was a required change. But I will move the cancel function higher up with a defer in the main.